core/src/doc/field.rs

```diff
@@ -69,6 +71,19 @@ impl Document {
 						},
 					}
 				}
+
+				// NONE-values should never be stored
+				if let Value::None = self.current.doc.pick(fd) {
```

Comment:
If know this is bike shedding but I feel like this is something Clippy might bring up in future.

```suggestion
				if self.current.doc.pick(fd).is_none() {
```